src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2443,6 +2443,18 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     GenerateImplicitConversionError(diagnostics, expression.Syntax, conversion, expression, targetType);
                 }
 
+                if (expression.Kind == BoundKind.TupleLiteral)
+                {
+                    // A BoundTupleLiteral must always be upgraded to a BoundConvertedTupleLiteral during conversion
+                    var sourceTuple = (BoundTupleLiteral)expression;
+                    expression = new BoundConvertedTupleLiteral(
```

Comment:
>expression = new BoundConvertedTupleLiteral( [](start = 20, length = 44)

I think it would be more robust to adjust SemanticModel implementation in Microsoft.CodeAnalysis.CSharp.CSharpSemanticModel.GetTypeInfoForNode